packages/dropdowns/src/Menu/Items/Item.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import StyledItem from '../../styled/items/StyledItem';
+import useDropdownContext from '../../utils/useDropdownContext';
+import useMenuContext from '../../utils/useMenuContext';
+
+// eslint-disable-next-line react/prop-types
+const Item = ({ value, disabled, component = StyledItem, ...props }) => {
+  const {
+    downshift: { selectedItem, highlightedIndex, getItemProps, setHighlightedIndex, itemToString }
+  } = useDropdownContext();
+  const { itemIndexRef } = useMenuContext();
+
+  if (!value && !disabled) {
+    throw new Error('All Item components require a `value` prop');
+  }
+
+  const currentIndex = itemIndexRef.current;
+  const isFocused = highlightedIndex === currentIndex;
+  let isSelected = selectedItem === value;
+
+  if (itemToString) {
+    isSelected = itemToString(selectedItem) === itemToString(value);
+  }
+
+  useEffect(() => {
+    if (!disabled && isSelected && highlightedIndex === null) {
+      setHighlightedIndex(currentIndex);
+    }
+  });
+
+  if (disabled) {
+    return React.createElement(component, {
+      disabled,
+      ...props
+    });
+  }
+
+  itemIndexRef.current++;
+
+  return React.createElement(
```

Comment:
Is there a reason for the sudden createElement usage?

Reply:
The `Item` component acts as the base for the `NextItem` and `PreviousItem` components.  The `.createElement()` usage lets us use all the logic within Item, but render with a different visual component.

```
// NextItem.js
<Item component={StyledNextItem} aria-expanded={true} value={value} {...props} />
```

There is also some additional logic that we track in the next and previous components to support the unique keyboard support.